src/Lykke.Service.B2c2Adapter/Services/OrderBooksService.cs

```diff
@@ -159,25 +190,74 @@ private OrderBook Convert(PriceMessage priceMessage)
             return result;
         }
 
+        private Task ReconnectIfNeeded(ITimerTrigger timer, TimerTriggeredHandlerArgs args, CancellationToken ct)
+        {
+            try
+            {
+                var isReceivedAtLeastOneOrderBookAtAll = _orderBooksCache.Values.Any();
+                var isNotReceivedAtLeastOneFreshOrderBookForTheLastInterval =
+                    !_orderBooksCache.Values.Any(x => DateTime.UtcNow - x.Timestamp < _reconnectIfNeededInterval);
+                var isNotConnectedFor1MinSinceStart = DateTime.UtcNow - Started > new TimeSpan(0, 0, 1, 0);
+
+                var needToReconnect = (isReceivedAtLeastOneOrderBookAtAll && isNotReceivedAtLeastOneFreshOrderBookForTheLastInterval)
+                    || (!isReceivedAtLeastOneOrderBookAtAll && isNotConnectedFor1MinSinceStart);
+
+                _log.Info($"Reconnect needed: {needToReconnect}. Order books count = {_orderBooksCache.Count}, " +
+                          $"received fresh message since last check = {!isNotReceivedAtLeastOneFreshOrderBookForTheLastInterval}, " +
+                          $"not connected for 1 min since start = {isNotConnectedFor1MinSinceStart}.");
+
+                if (needToReconnect)
+                    ForceReconnect();
+            }
+            catch (Exception ex)
```

Comment:
ForceReconnect already handles exceptions. Why do we need another one?

Reply:
fixed